tools/model-compiler/src/main/java/io/spine/tools/compiler/enrichment/Entry.java

```diff
@@ -20,116 +20,117 @@
+    Entry(DescriptorProto message, FieldDescriptorProto field, String packagePrefix) {
         this.message = message;
         this.field = field;
+        this.packagePrefix = packagePrefix;
     }
 
-    static boolean isSetFor(FieldDescriptorProto field) {
-        return Options.option(field, by)
-                      .isPresent();
+    String enrichmentType() {
+        String result = packagePrefix + message.getName();
+        return result;
     }
 
-    Map.Entry<String, String> collect(String packagePrefix) {
-        Collection<String> eventNames = parse();
-        Map.Entry<String, String> result = group(eventNames, packagePrefix);
+    String sourceTypes() {
+        Collection<String> sourceTypeNames = parse();
+        String result = group(sourceTypeNames);
         return result;
     }
 
     /**
-     * Obtains the list with fully-qualified names of target event types for
-     * the given field.
+     * Obtains the list with fully-qualified names of source event types for
+     * obtaining the value of the given enrichment field.
      */
     private List<String> parse() {
-        List<FieldReference> fieldRefs = FieldReference.allFrom(field);
+        List<FieldRef> fieldRefs = FieldRef.allFrom(field);
 
         ImmutableList.Builder<String> result = ImmutableList.builder();
-        for (FieldReference fieldRef : fieldRefs) {
+        for (FieldRef fieldRef : fieldRefs) {
             if (fieldRef.isWildcard() && fieldRefs.size() > 1) {
                 // Multiple argument `by` annotation can not contain wildcard reference onto
-                // the event type if the type was not specified with a `enrichment_for` annotation.
-                throw invalidByOptionUsage(field.getName());
+                // the source type if the type was not specified with a `enrichment_for` annotation.
+                throw invalidByOptionUsage();
             }
 
             if (fieldRef.isInner()) {
                 // The short form type names are handled as inner types.
                 continue;
             }
 
-            String fullTypeName = fieldRef.getType();
+            String fullTypeName = fieldRef.fullTypeName();
             result.add(fullTypeName);
         }
 
         return result.build();
     }
 
-    private Map.Entry<String, String> group(Collection<String> events, String packagePrefix) {
-        String enrichment = message.getName();
+    private String group(Collection<String> srcTypes) {
         String fieldName = field.getName();
-        Logger log = log();
-        Collection<String> eventGroup = new HashSet<>(events.size());
-        for (String eventName : events) {
-            if (eventName == null || eventName.trim()
-                                              .isEmpty()) {
-                throw invalidByOptionValue(enrichment);
+        List<String> list = new ArrayList<>();
+        for (String srcType : srcTypes) {
+            if (srcType == null || srcType.trim()
```

Comment:
You can use `Strings.isBlank()` here.

Reply:
I'd rather avoid dependency on `Roaster` just because of a string utility.